src/core/components/model.jsx

```diff
@@ -66,16 +66,20 @@ export default class Model extends ImmutablePureComponent {
     /*
      * If we have an unresolved ref, get the schema and name from the ref.
      * If the ref is external, we can't resolve it, so we just display the ref location.
-     * This is for situations where the ref was not resolved by Swagger Client
-     * because we reached the traversal depth limit.
+     * This is for situations where: 
+     *  - the ref was not resolved by Swagger Client because we reached the traversal depth limit
+     *  - we had a circular ref inside the allOf keyword
      */
     if ($ref) {
-      name = this.getModelName($ref)
-      const refSchema = this.getRefSchema(name)
+      const refName = this.getModelName($ref)
+      const refSchema = this.getRefSchema(refName)
       if (Map.isMap(refSchema)) {
-        schema = refSchema.set("$$ref", $ref)
-        $$ref = $ref
-      } else {
+        schema = refSchema.mergeDeep(schema) 
+        if (!$$ref) {
+          schema = schema.set("$$ref", $ref) 
+          $$ref = $ref
+        }
+      } else if (schema.size === 1) {
```

Comment:
This is in case there's no other property than `$ref` inside the schema and we didn't get a schema from `$ref`.

If there are other properties, then we have the situation with `allOf`, which means that we should still display the existing schema, even if it will be missing the properties from the unresolved `$ref`.